src/testdir/test_popup.vim

```diff
@@ -896,4 +896,48 @@ func Test_menu_only_exists_in_terminal()
   endtry
 endfunc
 
+func Test_popup_complete_mode()
+  new
+  inoremap <f5> <c-r>=complete_mode()<cr>
+  call writefile([
+        \ '!rm	src/testdir/amiga.vim	/^cmap !rm !Delete all$/;"	m',
+        \], 'dummy.txt')
+  setlocal tags=dummy.txt
+  setlocal dictionary=dummy.txt
+  setlocal thesaurus=dummy.txt
+  setlocal omnifunc=syntaxcomplete#Complete
+  setlocal completefunc=syntaxcomplete#Complete
+  setlocal spell
+  for [key, expected] in [
+        \ ["\<C-X>", 'ctrl_x'],
+        \ ["\<C-X>\<C-N>", 'keyword'],
+        \ ["\<C-X>\<C-L>", 'whole_line'],
+        \ ["\<C-X>\<C-F>", 'files'],
+        \ ["\<C-X>\<C-]>", 'tags'],
+        \ ["\<C-X>\<C-D>", 'path_defines'],
+        \ ["\<C-X>\<C-I>", 'path_patterns'],
+        \ ["\<C-X>\<C-K>", 'dictionary'],
+        \ ["\<C-X>\<C-T>", 'thesaurus'],
+        \ ["\<C-X>\<C-V>", 'cmdline'],
+        \ ["\<C-X>\<C-U>", 'function'],
+        \ ["\<C-X>\<C-O>", 'omni'],
+        \ ["\<C-X>s", 'spell'],
+        \]
+    call feedkeys("i" . key . "\<f5>\<ESC>", 'tx')
+    call assert_equal(expected, getline('.'))
+    %d
+  endfor
+  call delete('dummy.txt')
+
+  function! s:complTestEval() abort
```

Comment:
Vim tests no longer uses the exclamation mark in "function!" but uses "func" … "endfunc".